src/main/java/com/cs525/reversi/services/GameServiceImpl.java

```diff
@@ -30,41 +29,19 @@ public String createNewGame(NewGame newGameForm) {
 		User player1 = userRepo.findByUsername("comp");
 
 		if (player2 == null) {
-			player2 = new User();
-			player2.setUsername(newGameForm.getUserName());
-			player2.setUuid(UUID.randomUUID());
+			player2 = new User(UUID.randomUUID(), newGameForm.getUserName());
 			userRepo.save(player2);
 		}
 
-		Game game = new Game();
-		game.setPlayer1(player1);
-		game.setPlayer2(player2);
-		UUID gameUuid = UUID.randomUUID();
-		game.setUuid(gameUuid);
-		game.setRows(generateBoard());
-		game.setStatus(GameStatus.OPEN);
-		gameRepo.save(game);
-		return gameUuid.toString();
-	}
+		GameBuilder reversiGameBuilder = new ReversiGameBuilder();
 
-	private List<MatrixRow> generateBoard() {
-		List<MatrixRow> rows = new ArrayList<MatrixRow>();
+		Game game = reversiGameBuilder.buildPlayerOne(player1).buildPlayerTwo(player2).buildGameStatus(GameStatus.OPEN)
+				.buildGameUUID().buildBoardGame().getGame();
 
-		for (int i = 1; i <= 8; i++) {
-			MatrixRow row = new MatrixRow();
-			List<String> cells = new ArrayList<String>();
-			for (int j = 1; j <= 8; j++) {
-				if (i == 4 && j == 4 || i == 5 && j == 5)
-					cells.add("w");
-				else if (i == 4 && j == 5 || i == 5 && j == 4)
-					cells.add("b");
-				else
-					cells.add("");
-			}
-			row.setCells(cells);
-			rows.add(row);
-		}
-		return rows;
+		game.setDefaultCells();
```

Comment:
@MustafaMosad `setDefaultCells()` functionality could also be included inside the game builder and exposed via a `buildDefaultCells()` method